internal/ingress/annotations/annotations_test.go

```diff
@@ -27,21 +27,28 @@ import (
 
 func TestIngressClassValidatorFunc(t *testing.T) {
 	tests := []struct {
-		ingress              string
-		skipClasslessIngress bool
-		controller           string
-		isValid              bool
+		ingress              string // the class set on the Ingress resource
+		skipClasslessIngress bool   // the "user" classless ingress flag value
+		controller           string // the class set on the controller
+		isValid              bool   // the expected verdict
 	}{
 		{"", false, "", true},
-		{"", false, "kong", true},
-		{"", true, "kong", false},
-		{"kong", false, "kong", true},
-		{"kong", true, "kong", true},
+		{"", false, DefaultIngressClass, true},
+		{"", true, DefaultIngressClass, false},
+		{DefaultIngressClass, false, DefaultIngressClass, true},
+		{DefaultIngressClass, true, DefaultIngressClass, true},
 		{"custom", false, "custom", true},
-		{"", false, "killer", false},
-		{"custom", false, "kong", false},
-		{"custom", true, "kong", false},
+		{"", false, "killer", true},
+		{"custom", false, DefaultIngressClass, false},
+		{"custom", true, DefaultIngressClass, false},
 		{"", true, "custom", false},
+		{"", false, "kozel", true},
+		{"", true, "kozel", false},
+		{"kozel", false, "kozel", true},
+		{"kozel", true, "kozel", true},
+		{"", false, "killer", true},
+		{"custom", false, "kozel", false},
+		{"custom", true, "kozel", false},
```

Comment:
This table is bad. We should always use named struct parameters. Not in scope for this PR but leaving a note.